tests/misc_env/install_prereq.py

```diff
@@ -130,3 +132,26 @@ def enable_rhel_rpms(ceph, distro_ver):
     for repo in repos:
         ceph.exec_command(
             sudo=True, cmd='subscription-manager repos --enable={r}'.format(r=repo), long_running=True)
+
+
+def registry_login(ceph, distro_ver):
+    ''' login to this registry 'registry.redhat.io' on all nodes
+        docker for RHEL 7.x and podman for RHEL 8.x'''
+    cdn_cred = get_cephci_config().get('cdn_credentials')
+    if cdn_cred:
+        user = cdn_cred.get('username')
+        pwd = cdn_cred.get('password')
+        if not (user and pwd):
+            return
+
+        container = 'docker'
+        if distro_ver.startswith('8'):
+            container = 'podman'
+
+        ceph.exec_command(cmd='sudo yum install -y {c}'.format(c=container), long_running=True)
+
+        if container == 'docker':
+            ceph.exec_command(cmd='sudo systemctl restart docker', long_running=True)
+
+        ceph.exec_command(
+            cmd='sudo {c} login -u {u} -p {p} registry.redhat.io'.format(c=container, u=user, p=pwd), long_running=True)
```

Comment:
These look still redundant code for me. lot of `'if`s are present. you could try to use a dictionary here.

Reply:
@rakeshgm yes I agree with using dictionary where large items are there to check using 'if' but here two items also even using dictionary needs two checks also usage of dictionary makes slow down the performance of the code with so many opcode lookup operation, I feel it's ok with two 'if',  Please let me know your thoughts?